django/views/defaults.py

```diff
@@ -15,8 +17,21 @@ def page_not_found(request, template_name='404.html'):
     Context:
         request_path
             The path of the requested URL (e.g., '/app/pages/bad_page/')
+        exception
+            Any interesting message from the exception which triggered the 404,
+            or the exception class name
     """
-    context = {'request_path': request.path}
+    try:
```

Comment:
Does this version seem any better? (so we don't have to repeat the first statement two times)
```python
exception_repr = exception.__class__.__name__
try:
    message = exception.args[0]
except (AttributeError, IndexError):
    pass
else:
    if isinstance(message, six.text_type):
        exception_repr = message
```
Also, I'd add some comments to explain the purpose.